### PR TITLE
Refactor `ast::Stmt` and rename `ExprKind::Again` to `Continue`

### PR DESCRIPTION
**Pull request by jseyfried**
**syntax-[breaking-change] cc #31645**

Refactor away `ast::Decl`, refactor `ast::Stmt`, and rename `ast::ExprKind::Again` to `ast::ExprKind::Continue`.

r? @Manishearth

### CODE REVIEW

**Comment by Manishearth** [comment]
r? @eddyb 

(too many reviews right now)

File: src/libsyntax/ast.rs [hunk 1]
```diff
@@ -786,42 +785,38 @@ impl UnOp {
...
 
 
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash)]
 pub enum StmtKind {
-    /// Could be an item or a local (let) binding:
-    Decl(P<Decl>, NodeId),
+    /// A local (let) binding:
+    Local(P<Local>),
+
+    /// An item binding
```
Comment (eddyb) [inline src/libsyntax/ast.rs]: Nit: "An item definition".

**Comment by eddyb** [comment]
@Manishearth r=me with nit fixed.

**Comment by jseyfried** [comment]
Fixed the nit.

**Pull request merged**